presto-main/src/test/java/com/facebook/presto/server/TestParseToJsonData.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.server.protocol.QueryResourceUtil.parseToJson;
+import static org.testng.Assert.assertEquals;
+
+public class TestParseToJsonData
+{
+    @Test
```

Comment:
Can you add some tests, where the structural type contains key characters such as ":" and "{" to ensure they can be correctly escaped and parsed?

Reply:
I added 2 test cases of `assertNestedJson` to show json data can be serialized and deserialized correctly.